packages/common/src/contexts/wallet.tsx

```diff
@@ -1,241 +1,160 @@
-import {MathWalletAdapter} from "../wallet-adapters/mathWallet";
-
-const ASSETS_URL =
-  'https://raw.githubusercontent.com/solana-labs/oyster/main/assets/wallets/';
-export const WALLET_PROVIDERS = [
-  {
-    name: 'Phantom',
-    url: 'https://www.phantom.app',
-    icon: `https://www.phantom.app/img/logo.png`,
-    adapter: PhantomWalletAdapter,
-  },
-  {
-    name: 'Ledger',
-    url: 'https://www.ledger.com',
-    icon: `${ASSETS_URL}ledger.svg`,
-    adapter: LedgerWalletAdapter,
-  },
-  {
-    name: 'Sollet',
-    url: 'https://www.sollet.io',
-    icon: `${ASSETS_URL}sollet.svg`,
-  },
-  {
-    name: 'Solong',
-    url: 'https://solongwallet.com',
-    icon: `${ASSETS_URL}solong.png`,
-    adapter: SolongWalletAdapter,
-  },
-  // TODO: enable when fully functional
-  {
-    name: 'MathWallet',
-    url: 'https://mathwallet.org',
-    icon: `${ASSETS_URL}mathwallet.svg`,
-    adapter: MathWalletAdapter,
-  },
-  // {
-  //   name: 'Torus',
-  //   url: 'https://tor.us',
-  //   icon: `${ASSETS_URL}torus.svg`,
-  //   adapter: TorusWalletAdapter,
-  // }
-
-  // Solflare doesnt allow external connections for all apps
-  // {
-  //   name: "Solflare",
-  //   url: "https://solflare.com/access-wallet",
-  //   icon: `${ASSETS_URL}solflare.svg`,
-  // },
-];
+import { WalletAdapter, WalletError } from '@solana/wallet-adapter-base';
+import { useWallet, WalletProvider as BaseWalletProvider } from '@solana/wallet-adapter-react';
+import {
+  getLedgerWallet,
+  getMathWallet,
+  getPhantomWallet,
+  getSolflareWallet,
+  getSolletWallet,
+  getSolongWallet,
+  getTorusWallet,
+} from '@solana/wallet-adapter-wallets';
+import { Button, Modal } from "antd";
+import React, { createContext, FC, ReactNode, useCallback, useContext, useEffect, useMemo, useState } from 'react';
+import { notify } from "../utils";
+
+export interface WalletModalContextState {
+  visible: boolean;
+  setVisible: (open: boolean) => void;
+}
 
-const WalletContext = React.createContext<{
-  wallet: WalletAdapter | undefined;
-  connected: boolean;
-  select: () => void;
-  provider: typeof WALLET_PROVIDERS[number] | undefined;
-}>({
-  wallet: undefined,
-  connected: false,
-  select() {},
-  provider: undefined,
-});
+export const WalletModalContext = createContext<WalletModalContextState>({} as WalletModalContextState);
 
-export function WalletProvider({ children = null as any }) {
-  const { endpoint } = useConnectionConfig();
-  const location = useLocation();
-  const [autoConnect, setAutoConnect] = useState(
-    location.pathname.indexOf('result=') >= 0 || false,
-  );
-  const [providerUrl, setProviderUrl] = useLocalStorageState('walletProvider');
+export function useWalletModal(): WalletModalContextState {
+  return useContext(WalletModalContext);
+}
 
-  const provider = useMemo(
-    () => WALLET_PROVIDERS.find(({ url }) => url === providerUrl),
-    [providerUrl],
-  );
+export const WalletModal: FC = () => {
```

Comment:
nit: We don't quite have consistency in the repo :) but I would prefer to just use function here and avoid explicit type specification.

Reply:
Your call, though I think the type is helpful in case a component fails to return a valid `ReactNode` or `null` value. What's the reason to not have them?

Reply:
Mainly personal preferences. In general I prefer when types are inferred rather than being specified explicitly. There are also small technical nuances. For example the FC component would have children prop even though it doesn't use them. 

Regarding the result check without the explicit type if it returned an invalid value then you would know as soon as you would try to use it, so I think it's not that important here.

Reply:
Yeah, makes sense. I'll remove here then.